src/channel.ts

```diff
@@ -787,6 +788,43 @@ export class Channel<StreamChatGenerics extends ExtendableGenerics = DefaultGene
     }
   }
 
+  /**
+   * Sends an event to update the AI state for a specific message.
+   * Typically used by the server connected to the AI service to notify clients of state changes.
+   *
+   * @param messageId - The ID of the message associated with the AI state.
+   * @param state - The new state of the AI process (e.g., thinking, generating).
+   * @param options - Optional parameters, such as `ai_message`, to include additional details in the event.
+   */
+  async updateAIState(messageId: string, state: AIState, options?: { ai_message: string }) {
+    await this.sendEvent({
+      type: 'ai_indicator.update',
+      message_id: messageId,
+      ai_state: state,
+      ...(options || {}),
+    } as Event<StreamChatGenerics>);
+  }
```

Comment:
```suggestion
  async updateAIState(messageId: string, state: AIState, options: { ai_message: string } = {}) {
    await this.sendEvent({
      ...options,
      type: 'ai_indicator.update',
      message_id: messageId,
      ai_state: state,
    } as Event<StreamChatGenerics>);
  }
```

Unless we want to allow full override of the object passed to the send event, I'd move option spread to the top (JS users do not have/rely on types).

Reply:
Thanks for the feedback, @arnautov-anton ❤️